tests/components/velbus/test_entity.py

```diff
@@ -0,0 +1,50 @@
+"""Velbus entity tests."""
+
+from unittest.mock import AsyncMock
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.const import ATTR_ENTITY_ID, SERVICE_TURN_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry as dr
+
+from . import init_integration
+
+from tests.common import MockConfigEntry
+
+
+async def test_api_call(
+    hass: HomeAssistant,
+    mock_relay: AsyncMock,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test the api call decorator action."""
+    await init_integration(hass, config_entry)
+
+    mock_relay.turn_on.side_effect = OSError()
+    with pytest.raises(HomeAssistantError):
+        await hass.services.async_call(
+            SWITCH_DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: "switch.relayname"},
+            blocking=True,
+        )
+
+
+async def test_device_registry(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    device_registry: dr.DeviceRegistry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the velbus device registry."""
+    await init_integration(hass, config_entry)
+
+    # Ensure devices are correctly registered
+    device_entries = dr.async_entries_for_config_entry(
+        device_registry, config_entry.entry_id
+    )
+    assert device_entries == snapshot
```

Comment:
I would propose to also check some devices with hard asserts to make sure the device are connected to the right one (since the snapshot does `'via_device_id': <ANY>,`)